src/Compilers/Shared/RuntimeHostInfo.cs

```diff
@@ -49,40 +49,15 @@ internal static (string processFilePath, string commandLineArguments, string too
 
         internal static bool IsCoreClrRuntime => true;
 
+        private const string DotNetHostPathEnvironmentName = "DOTNET_HOST_PATH";
+
         /// <summary>
-        /// Get the path to the dotnet executable. In the case the host did not provide this information
+        /// Get the path to the dotnet executable. In the case the .NET SDK did not provide this information
         /// in the environment this will return simply "dotnet".
         /// </summary>
-        /// <remarks>
-        /// See the following issue for rationale why only %PATH% is considered
-        /// https://github.com/dotnet/runtime/issues/88754
-        /// </remarks>
-        internal static string GetDotNetPathOrDefault()
-        {
-            var (fileName, sep) = PlatformInformation.IsWindows
-                ? ("dotnet.exe", ';')
-                : ("dotnet", ':');
-
-            var path = Environment.GetEnvironmentVariable("PATH") ?? "";
```

Comment:
I would prefer to make this change smaller, until @jaredpar gets back. I'm not entirely sure why he manually searched the entire `PATH` for `dotnet`, as opposed to just doing what this change proposes, and if we want to get this in for 8p7 we don't have time to wait to ask. Let's do the minimal change required, which is to check `DOTNET_HOST_PATH`. If that exists and points to a valid thing, use that, otherwise do what the current code is doing.

Reply:
I've taken this approach. There is a very high probably this won't be the long-term approach, but it addresses your concern for a minimal change for now while unblocking the scenario.